rocks-db/src/batch_client.rs

```diff
@@ -142,13 +146,29 @@ impl AssetIndexReader for Storage {
             .iter()
             .flat_map(|c| c.as_ref().map(|c| c.collection.value))
             .collect::<Vec<_>>();
```

Comment:
You need to collect those pks only when you don't have the collection_authorities passed in.